### PR TITLE
[Bugfix] ListView: horizontal grid 

### PR DESCRIPTION
**Pull request by ldanilina**
Horizontal grids fixed for TopHorizontal and BottomHorizontal layouts.

image

### CODE REVIEW

File: LocalPackages/DolbyIORTSUIKit/Sources/DolbyIORTSUIKit/Private/Views/ListView.swift [hunk 1]
```diff
@@ -248,6 +245,30 @@ struct ListView: View {
         }
     }
 
+    private func gridHorizontal(availableHeight: CGFloat) -> ForEach<[StreamSource], UUID, HStack<(some View)?>> {
```
Comment (aravind-raveendran) [inline LocalPackages/DolbyIORTSUIKit/Sources/DolbyIORTSUIKit/Private/Views/ListView.swift:248]: Did you see if we can return `LazyVGrid` or `LazyHGrid` from these functions so the return type can actually be a bit simpler  - `some View` ?

File: LocalPackages/DolbyIORTSUIKit/Sources/DolbyIORTSUIKit/Private/Views/ListView.swift [hunk 2]
```diff
@@ -95,17 +95,15 @@ struct ListView: View {
...
     private func topHorizontalLayout(_ screenSize: CGSize, _ rowsCount: Int) -> some View {
         VStack {
+            let maxAllowedMainVideoSize = CGSize(width: screenSize.width, height: screenSize.height * Defaults.maximumNumberOfTilesRatio)
             if let source = viewModel.selectedVideoSource, let mainViewProvider = viewModel.mainViewProvider(for: source) {
-                let maxAllowedMainVideoSize = CGSize(width: screenSize.width, height: screenSize.height * Defaults.maximumNumberOfTilesRatio)
                 mainView(screenSize, mainViewProvider, source, maxAllowedMainVideoSize)
             }
             ScrollView(.horizontal) {
-                let rows = [GridItem](repeating: GridItem(.flexible(), spacing: Layout.spacing1x), count: rowsCount)
+                let availableHeight = (screenSize.height - maxAllowedMainVideoSize.height) / (rowsCount <= 4 ? 4 :  CGFloat(rowsCount))
```
Comment (CasperMok) [inline LocalPackages/DolbyIORTSUIKit/Sources/DolbyIORTSUIKit/Private/Views/ListView.swift:103]: Just the location difference from bottomHorizontalLayout(), please use a similar algorithm to do the width/height in both, may use a common function. This ensure the similar look and feel on both of them.
Please see if a similar shall be done on both topVerticalLayout and bottomVerticalLayout.

Comment (ldanilina) [reply LocalPackages/DolbyIORTSUIKit/Sources/DolbyIORTSUIKit/Private/Views/ListView.swift]: fixed, thanks

File: LocalPackages/DolbyIORTSUIKit/Sources/DolbyIORTSUIKit/Private/Views/ListView.swift (hunk 1 above)
Comment (ldanilina) [reply LocalPackages/DolbyIORTSUIKit/Sources/DolbyIORTSUIKit/Private/Views/ListView.swift:248]: we can't, different configurations of LazyxGrid is used for each case..

**Comment by CasperMok** [comment]
Now some of the layout functions look very similar, such as topVerticalLayout() and bottomVerticalLayout(). I suspect we can further simplify/optimize the code. Let's do it later

**Review by CasperMok — APPROVED** [review]

**Pull request merged**